### PR TITLE
Refactor cuda adapter to new logger

### PR DESCRIPTION
**Pull request by lplewa**

### CODE REVIEW

**Comment by lplewa** [comment]
I created an octopus merge b45470f9e1a5c4bacbc210d68e900858b82fae25 of all logs changes (#1359; #1347; #1345; #1342; #1032) and tested integration with [intel/llvm repository](https://github.com/intel/llvm/pull/12580/files)

**Comment by lplewa** [comment]
@oneapi-src/unified-runtime-cuda-write - please review - if you are ok with changes please label it as ready to merge

File: source/adapters/cuda/adapter.cpp [hunk 1]
```diff
@@ -11,14 +11,42 @@
...
+                     const std::string &msg) override {
+    std::cerr << msg << std::endl;
+  }
+
+  ~ur_legacy_sink() = default;
+};
+ur_adapter_handle_t_::ur_adapter_handle_t_()
+    : logger(logger::get_logger("cuda")) {
+
+  if (std::getenv("UR_LOG_CUDA") != NULL)
```
Comment (npmiller) [inline source/adapters/cuda/adapter.cpp:42]:
```suggestion
  if (std::getenv("UR_LOG_CUDA") != nullptr)
```

Aligning this with the `nullptr` use below

File: source/adapters/cuda/command_buffer.hpp [hunk 2]
```diff
@@ -169,10 +170,10 @@ static inline const char *getUrResultString(ur_result_t Result) {
 #define UR_CALL(Call, Result)                                                  \
   {                                                                            \
     if (PrintTrace)                                                            \
-      fprintf(stderr, "UR ---> %s\n", #Call);                                  \
+      logger::always("UR ---> {}", #Call);                                     \
     Result = (Call);                                                           \
     if (PrintTrace)                                                            \
-      fprintf(stderr, "UR <--- %s(%s)\n", #Call, getUrResultString(Result));   \
+      logger::always("UR <--- {}({})\n", #Call, getUrResultString(Result));    \
```
Comment (Bensuo) [inline source/adapters/cuda/command_buffer.hpp:176]: Should `\n` have been removed here? Or should it also have been removed on line 176?

Comment (lplewa) [reply source/adapters/cuda/command_buffer.hpp]: done

File: source/adapters/cuda/adapter.cpp (hunk 1 above)
Comment (lplewa) [reply source/adapters/cuda/adapter.cpp]: done

**Review by npmiller — APPROVED** [review]

**Pull request merged**